universal-application-tool-0.0.1/test/controllers/admin/AdminQuestionControllerTest.java

```diff
@@ -245,6 +246,36 @@ public void update_redirectsOnSuccess() {
     assertThat(result.flash().get("message").get()).contains("updated");
   }
 
+  @Test
+  public void update_setsIdsAsExpected() {
+    DropdownQuestionDefinition definition =
+        new DropdownQuestionDefinition(
+            /* name= */ "applicant ice cream",
+            /* enumeratorId= */ Optional.empty(),
+            /* description= */ "Select your favorite ice cream flavor",
+            LocalizedStrings.of(Locale.US, "Ice cream?", Locale.FRENCH, "crème glacée?"),
+            LocalizedStrings.of(Locale.US, "help", Locale.FRENCH, "aider"),
+            ImmutableList.of(
+                QuestionOption.create(
+                    1L, LocalizedStrings.of(Locale.US, "chocolate", Locale.FRENCH, "chocolat")),
+                QuestionOption.create(
+                    2L, LocalizedStrings.of(Locale.US, "strawberry", Locale.FRENCH, "fraise")),
+                QuestionOption.create(
+                    3L, LocalizedStrings.of(Locale.US, "vanilla", Locale.FRENCH, "vanille")),
+                QuestionOption.create(
+                    4L, LocalizedStrings.of(Locale.US, "coffee", Locale.FRENCH, "café"))));
+    // We can only update draft questions, so save this in the DRAFT version.
+    testQuestionBank.maybeSave(definition, LifecycleStage.DRAFT);
+
+    DropdownQuestionForm questionForm = new DropdownQuestionForm(definition);
+    questionForm.setNewOptions(ImmutableList.of("cookie", "mint", "pistachio"));
```

Comment:
It'd be good to check that these get the correct IDs too, and not just that another field in the data is updated.

Reply:
Added, let me know what you think!

Reply:
Okay, this made me realize we need to update the options IDs list when adding new options. Just wanted to call attention to that line in MultiOptionQuestionForm -- let me know if it doesn't look right to you

Reply:
Please ignore that previous comment, I was wrong. Tests should all be passing and code should be working as expected!